traffic_monitor/health/cache.go

```diff
@@ -162,45 +163,90 @@ func EvalCache(result cache.ResultInfo, resultStats cache.ResultStatValHistory,
 		}
 
 		if !inThreshold(threshold, resultStatNum) {
-			return false, eventDesc(status, exceedsThresholdMsg(stat, threshold, resultStatNum)), stat
+			return false, result.UsingIPV4, eventDesc(status, exceedsThresholdMsg(stat, threshold, resultStatNum)), stat
 		}
 	}
 
-	return result.Available, eventDesc(status, availability), ""
+	return result.Available, result.UsingIPV4, eventDesc(status, availability), ""
 }
 
 // CalcAvailability calculates the availability of the cache, from the given result. Availability is stored in `localCacheStatus` and `localStates`, and if the status changed an event is added to `events`. statResultHistory may be nil, for pollers which don't poll stats.
 // TODO add tc for poller names?
-func CalcAvailability(results []cache.Result, pollerName string, statResultHistory cache.ResultStatHistory, mc tc.TrafficMonitorConfigMap, toData todata.TOData, localCacheStatusThreadsafe threadsafe.CacheAvailableStatus, localStates peer.CRStatesThreadsafe, events ThreadsafeEvents) {
+func CalcAvailability(results []cache.Result, pollerName string, statResultHistory cache.ResultStatHistory, mc tc.TrafficMonitorConfigMap, toData todata.TOData, localCacheStatusThreadsafe threadsafe.CacheAvailableStatus, localStates peer.CRStatesThreadsafe, events ThreadsafeEvents, protocol config.PollingProtocol) {
 	localCacheStatuses := localCacheStatusThreadsafe.Get().Copy()
+	processAvailableTuple := func(tuple cache.AvailableTuple) bool {
+		switch protocol {
+		case config.IPV4Only:
+			return tuple.IPV4
+			break
+		case config.IPV6Only:
+			return tuple.IPV6
+			break
+		case config.Both:
+			return tuple.IPV4 && tuple.IPV6
+			break
+		}
```

Comment:
This needs a `default` case, which logs an error that the `protocol` was unexpected. (Probably still correct to mark the cache as offline.) If you make the `PollingProtocol` `Unmarshal` func return an error like I suggested, that should never happen, but it should still log an error if it does, as a failsafe.